lntest/itest/lnd_test.go

```diff
@@ -4406,43 +4405,135 @@ func testMultiHopPayments(net *lntest.NetworkHarness, t *harnessTest) {
 	closeChannelAndAssert(ctxt, t, net, carol, chanPointCarol, false)
 }
 
-// testSingleHopSendToRoute tests that payments are properly processed
-// through a provided route with a single hop. We'll create the
-// following network topology:
-//      Alice --100k--> Bob
-// We'll query the daemon for routes from Alice to Bob and then
-// send payments through the route.
+type singleHopSendToRouteCase struct {
+	name string
+
+	// streaming tests streaming SendToRoute if true, otherwise tests
+	// synchronous SenToRoute.
```

Comment:
Hm, when is the time to get rid of one of these RPCs?

Reply:
i wouldn't be opposed to deprecating the streaming version and telling people to switch to `SendToRouteSync`, or `RouterRPC.SendToRoute` if they don't mind dealing with breaking changes.